invenio_vocabularies/records/systemfields/pid.py

```diff
@@ -93,10 +89,26 @@ def resolve(self, pid_value):
 
         return record
 
+    def get_pid_type(self, type_id):
+        """Get the PID type for a vocabulary type."""
+        # Get type based on name.
+        vocab_type = VocabularyType.query.filter_by(id=type_id).one_or_none()
+        if vocab_type is None:
+            raise PIDDoesNotExistError(None, None)
+        return vocab_type.pid_type
+
     @property
-    def type_id(self):
+    def pid_type(self):
         """Get the current defined type."""
-        return getattr(self, '_type_id', None)
+        # This ensures that when we use Vocabulary.pid.with_type_ctx('...')
+        # we cache the pid type to avoid querying the database every time.
+        type_id = getattr(self, '_type_id', None)
```

Comment:
where would `_type_id` be set? I cannot find it so Im thinking it would always be None? in most cases it was changes by `_pid_type`, which is set below in line 110...